src/main/java/org/truffleruby/core/kernel/KernelNodes.java

```diff
@@ -992,6 +998,15 @@ private boolean isLiteralBlock(Node callNode) {
             return false;
         }
 
+        private void warnProcWithoutBlock() {
+            if (warnNode == null) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
+                warnNode = insert(new WarnNode());
+            }
+            final SourceSection sourceSection = getContext().getCallStack().getCallerFrameIgnoringSend().getCallNode().getEncapsulatingSourceSection();
```

Comment:
This is fairly long, could we have a helper like CallStackManager#getTopMostUserSourceSection(), like getTopMostUserCallNode() but returning a SourceSection?

Reply:
Added, please review.